src/core/qgsapplication.cpp

```diff
@@ -1095,7 +1095,8 @@ QString QgsApplication::userStylePath()
 
 QRegExp QgsApplication::shortNameRegExp()
 {
-  return QRegExp( "^[A-Za-z][A-Za-z0-9\\._-]*" );
+  static const QRegExp regexp( QStringLiteral( "^[A-Za-z][A-Za-z0-9\\._-]*" ) );
```

Comment:
Does changing this really improve performanc? Or would it be the same as marking the function static const?

Reply:
The function could be static which is a good idea, but doesn't cache the value automatically.
Making a function static const is not possible afaict.

Reply:
Thanks!

Reply:
Actually it is already static